lib/boxmaker.scad

```diff
@@ -0,0 +1,176 @@
+include <flat_pack.scad>;
+
+/*
+ * A finger-jointed lasercut box maker OpenSCAD library.
+ *
+ * Inspired by http://boxdesigner.connectionlab.org/
+ */
+module box_2d(box_inner, thickness, tabs, margin=2) {
+  layout_2d(box_inner, thickness) {
+    side_a_top(box_inner, thickness, tabs);
+    side_a(box_inner, thickness, tabs);
+    side_b(box_inner, thickness, tabs);
+    side_b(box_inner, thickness, tabs);
+    side_c(box_inner, thickness, tabs);
+    side_c(box_inner, thickness, tabs);
+  }
+}
+
+// An assembled version of the box for previewing.
+module box_3d(box_inner, thickness, tabs) {
+  layout_3d(box_inner, thickness) {
+    side_a_top(box_inner, thickness, tabs);
+    side_a(box_inner, thickness, tabs);
+    side_b(box_inner, thickness, tabs);
+    side_b(box_inner, thickness, tabs);
+    side_c(box_inner, thickness, tabs);
+    side_c(box_inner, thickness, tabs);
+  }
+}
+
+// lasercut layout
+module layout_2d(inner, thickness, margin=2) {
+  spacingA = inner[0] + thickness * 2 + margin;
+  spacingB = inner[1] + thickness * 2 + margin;
+
+  // bottom
+  children(0);
+  // top
+  translate([spacingA, 0, 0])
+    children(1);
+  // right
+  translate([spacingA * 2, inner[1], 0])
+    rotate([0, 0, -90])
+      children(2);
+  // left
+  translate([spacingA * 2 + inner[2] + thickness * 2 + margin, inner[1], 0])
+    rotate([0, 0, -90])
+      children(3);
+  // front
+  translate([0, inner[1] + margin + thickness * 2, 0])
+    children(4);
+  // back
+  translate([inner[0] + margin + thickness * 2,
+             inner[1] + margin + thickness * 2, 0])
+    children(5);
+}
+
+// An assembled version of the box for previewing.
+module layout_3d(box_inner, thickness) {
```

Comment:
This looks good; is this only for visualization of the laser cutting, or also for 3D printing? This module probably won't produce usable 3D printed parts due to thermal expansion.

Reply:
Yeah, probably worth mentioning somewhere that 3D printing would require adding in some kerf values that don't exist.

Was it your module that attempted to add kerf handling, then reverted it because it added unnecessary complexity and laser cutter kerf works out well enough for making pressure-fit + glue joints?

Reply:
I initially added kerf calculations in, but then realized in a comb joint with the same sized cuts, the kerf worked its self out and ends up being split between the A and B faces. For a 3D printed face the math is a little harder. I think that one face or the other will need to take into account the size difference. 

That throws all the math out a bit. If we want to add that as a feature it'll need to be added into the outside and inside cuts library.